src/list.rs

```diff
@@ -32,8 +33,47 @@ fn handle_list(app_state: &mut AppState, stdout: &mut StdoutLock) -> Result<()>
 
                 list_state.message.clear();
 
+                let curr_key = key.code;
+
+                if is_searching {
+                    match curr_key {
+                        KeyCode::Esc | KeyCode::Enter => {
+                            is_searching = false;
+                            list_state.search_query.clear();
+                            continue;
+                        }
+                        KeyCode::Char(k) => {
+                            list_state.search_query.push(k);
+                            list_state.message.push_str("search:");
+                            list_state.message.push_str(&list_state.search_query);
+                            list_state.message.push('|');
+
+                            list_state.select_if_matches_search_query();
```

Comment:
Updating the message and then calling `select_if_matches_search_query` is duplicated. It would be better to replace `select_if_matches_search_query` with something like `apply_search_query` which also updates the message.

Reply:
made the refactor!